internal/cmdrender/executor.go

```diff
@@ -429,31 +421,38 @@ func cloneResources(input []*yaml.RNode) (output []*yaml.RNode) {
 }
 
 // path (location) of a KRM resources is tracked in a special key in
-// metadata.annotation field. adjustRelPath updates that path annotation by prepending
-// the given relPath to the current path annotation if it doesn't exist already.
-// Resources are read from local filesystem or generated at a package level, so the
-// path annotation in each resource points to path relative to that package.
-// But the resources are written to the file system at the root package level, so
+// metadata.annotation field that is used to write the resources to the filesystem.
+// When resources are read from local filesystem or generated at a package level, the
+// path annotation in a resource points to path relative to that package. But the resources
+// are written to the file system at the root package level, so
 // the path annotation in each resources needs to be adjusted to be relative to the rootPkg.
-func adjustRelPath(resources []*yaml.RNode, relPath string) ([]*yaml.RNode, error) {
-	if relPath == "" {
-		return resources, nil
-	}
+// adjustRelPath updates the path annotation by prepending the path of the package
+// a resource relative to the root package.
+func adjustRelPath(resources []*yaml.RNode, rootPath types.UniquePath) error {
```

Comment:
This function signature is at the wrong abstractions. Can you extract a pure function that only deals with Paths. That function should not know anything about YAML AST. It should then be unit tested.

Reply:
Good point. Abstracted out the relative path computation logic in a separate function and added test-cases.

Reply:
Also added additional `e2e` test cases for `fn-render` and `fn-eval`.